src/core/libraries/rtc/rtc.cpp

```diff
@@ -5,154 +5,397 @@
+int PS4_SYSV_ABI sceRtcFormatRFC3339Precise(char* pszDateTime, const OrbisRtcTick* pTickUtc,
+                                            int minutes) {
     LOG_ERROR(Lib_Rtc, "(STUBBED) called");
     return ORBIS_OK;
 }
 
-int PS4_SYSV_ABI sceRtcFormatRFC3339PreciseLocalTime() {
+int PS4_SYSV_ABI sceRtcFormatRFC3339PreciseLocalTime(char* pszDateTime,
+                                                     const OrbisRtcTick* pTickUtc) {
     LOG_ERROR(Lib_Rtc, "(STUBBED) called");
     return ORBIS_OK;
 }
 
-int PS4_SYSV_ABI sceRtcGetCurrentAdNetworkTick() {
-    LOG_ERROR(Lib_Rtc, "(STUBBED) called");
-    return ORBIS_OK;
+int PS4_SYSV_ABI sceRtcGetCurrentAdNetworkTick(OrbisRtcTick* pTick) {
+    LOG_TRACE(Lib_Rtc, "called");
+
+    if (pTick == nullptr)
+        return ORBIS_RTC_ERROR_INVALID_POINTER;
+
+    Kernel::OrbisKernelTimespec clocktime;
+    int returnValue = Kernel::sceKernelClockGettime(Kernel::ORBIS_CLOCK_EXT_NETWORK, &clocktime);
+
+    if (returnValue == 0) {
+        pTick->tick = clocktime.tv_nsec / 1000 + clocktime.tv_sec * 1000000 + 0xdcbffeff2bc000;
+    } else {
+        return ORBIS_RTC_ERROR_NOT_INITIALIZED;
+    }
+
+    return 0;
 }
 
-int PS4_SYSV_ABI sceRtcGetCurrentClock() {
-    LOG_ERROR(Lib_Rtc, "(STUBBED) called");
-    return ORBIS_OK;
+int PS4_SYSV_ABI sceRtcGetCurrentClock(OrbisRtcDateTime* pTime, int timeZone) {
+    LOG_TRACE(Lib_Rtc, "called");
+
+    if (pTime == nullptr)
+        return ORBIS_RTC_ERROR_INVALID_POINTER;
+
+    Kernel::OrbisKernelTimespec clocktime;
+    int returnValue = Kernel::sceKernelClockGettime(Kernel::ORBIS_CLOCK_REALTIME, &clocktime);
+
+    if (returnValue == 0) {
+        OrbisRtcTick clockTick;
+        clockTick.tick = clocktime.tv_nsec / 1000 + clocktime.tv_sec * 1000000 + 0xdcbffeff2bc000;
+
+        sceRtcTickAddMinutes(&clockTick, &clockTick, timeZone);
+        sceRtcSetTick(pTime, &clockTick);
+    }
+
+    return returnValue;
 }
 
-int PS4_SYSV_ABI sceRtcGetCurrentClockLocalTime() {
-    LOG_ERROR(Lib_Rtc, "(STUBBED) called");
-    return ORBIS_OK;
+int PS4_SYSV_ABI sceRtcGetCurrentClockLocalTime(OrbisRtcDateTime* pTime) {
+    LOG_TRACE(Lib_Rtc, "called");
+
+    if (pTime == nullptr)
+        return ORBIS_RTC_ERROR_INVALID_POINTER;
+
+    Kernel::OrbisKernelTimespec clocktime;
+    int returnValue = Kernel::sceKernelClockGettime(Kernel::ORBIS_CLOCK_REALTIME, &clocktime);
+
+    if (returnValue >= 0) {
+        u64 clockTick = clocktime.tv_nsec / 1000 + clocktime.tv_sec * 1000000 + 0xdcbffeff2bc000;
+
+        time_t seconds;
+        Kernel::OrbisTimesec timezone;
+
+        returnValue =
+            Kernel::sceKernelConvertUtcToLocaltime(clockTick / 1000000, &seconds, &timezone, 0);
+
+        if (returnValue >= 0) {
+            OrbisRtcTick newTick;
+            newTick.tick = clockTick;
+            sceRtcTickAddMinutes(&newTick, &newTick, (timezone.dst_sec + timezone.west_sec) / 0x3c);
+            sceRtcSetTick(pTime, &newTick);
+        }
```

Comment:
Replace all inside main if block above with this, `clockTick` calculation is not necessary, and simplifying this `newTick` calculation can cause floating point imprecision. There can be games which use this for update loops, so this is problematic to simplify. There may be some OS-specific problems here, so will need to test.
```suggestion
        tzOffset = timeZone.tz_minuteswest + timeZone.tz_dsttime; // calculate total timezone offset for converting UTC to local time
        returnValue = sceKernelClockGettime(0, &clocktime);
        if (returnValue >= 0)
                OrbisRtcTick newTick;
                newTick.tick = ((unsigned __int64)((unsigned __int128)(clocktime.tv_nsec * (signed __int128)2361183241434822607LL) >> 64) >> 63)
                + ((signed __int64)((unsigned __int128)(clocktime.tv_nsec * (signed __int128)2361183241434822607LL) >> 64) >> 7)
                + 1000000 * clocktime.tv_sec
                + 62135596800000000LL;
        sceRtcTickAddMinutes(&newTick, &newTick, tzOffset);
        sceRtcSetTick(pTime, &newTick);
```